functions/Fetch_JSON_Data_from_PublicApi.js

```diff
@@ -0,0 +1,88 @@
+const https = require('https');
+const url = require('url');
+
+exports.handler = (request, context, callback) => {
+    update(request.state, request.secrets, callback);
+};
+
+function update(state, secret, callback) {
+    let result = initializeState(state);
+    let since = state.astroidsCursor || "2018-05-01";
+    let baseApi = "https://api.nasa.gov/neo/rest/v1/feed?start_date=" + new Date(since).toISOString().slice(0,10) + "&api_key=" + secret.apiKey;
+        const request = https.get(baseApi, response => {
+            if (response.statusCode < 200 || response.statusCode > 299) {
+                callback(new Error('Request failed with code: ' + response.statusCode), null);
```

Comment:
from what I saw in other examples, we just omit second parameter in case of error.
Not sure what's the right way here, but we should be consistent.

Reply:
I think we should pass both the parameters to be consistent with other `callback()` method calls. Also it will be explicit that `callback()` takes two arguments, one error and another response.

Reply:
Right. will change it. For the sake of clarity, we can add it back in all samples in code cleanup.